BMM.Core/GuardedActions/Player/UpdateExternalRelationsAction.cs

```diff
@@ -36,19 +38,33 @@ protected override Task Execute()
             PlayerViewModel.HasExternalRelations = currentTrack.Relations != null &&
                                                    currentTrack.Relations.Any(relation => relation.Type == TrackRelationType.External);
 
+            PlayerViewModel.SongTreasureLink =  GetLyricsLink(currentTrack);
+            return Task.CompletedTask;
+        }
+
+        private string GetLyricsLink(ITrackModel currentTrack)
+        {
             var existingSongbook = currentTrack
                 ?.Relations
                 ?.OfType<TrackRelationSongbook>()
                 .FirstOrDefault();
 
-            string songTreasureLink = existingSongbook != null
-                ? string.Format(_firebaseRemoteConfig.SongTreasuresSongLink,
+            if (existingSongbook != null)
+            {
+                return string.Format(_firebaseRemoteConfig.SongTreasuresSongLink,
                     SongbookUtils.GetShortName(existingSongbook.Name),
-                    existingSongbook.Id)
-                : string.Empty;
-            
-            PlayerViewModel.SongTreasureLink = songTreasureLink;
-            return Task.CompletedTask;
+                    existingSongbook.Id);
+            }
+
+            var sangtekstElement = currentTrack
+                ?.Relations
+                ?.OfType<TrackRelationExternal>()
+                .FirstOrDefault(x => x.Name == SangtekstRelationName);
```

Comment:
the PBI states "(ignoring upper or smaller case)"

Reply:
my oversight, sorry